From e9053d311e689e937bfdb451f872627073af4ac6 Mon Sep 17 00:00:00 2001
From: Gerry Kopec <gerry.kopec@windriver.com>
Date: Tue, 2 Oct 2018 03:04:26 +0000
Subject: [PATCH 2/3] Revert "Openstack: Use k8s secret to store config"

This reverts commit 83b91e6e1be4bcab83c333f993fdc9e6fa8fa4bc.

Conflicts:
	gnocchi/templates/configmap-etc.yaml
	gnocchi/templates/daemonset-metricd.yaml
	gnocchi/templates/daemonset-statsd.yaml
	gnocchi/templates/deployment-api.yaml
	gnocchi/templates/job-db-init-indexer.yaml
	gnocchi/templates/job-db-sync.yaml
	gnocchi/templates/pod-gnocchi-test.yaml

To allow per host overrides to work we need to temporarily revert this
commit.  This should be put back once upstream has fixed the issue:
https://storyboard.openstack.org/#!/story/2003873
---
 barbican/templates/configmap-etc.yaml              | 21 +++++---
 barbican/templates/deployment-api.yaml             |  4 +-
 ceilometer/templates/configmap-etc.yaml            | 29 ++++++-----
 ceilometer/templates/daemonset-compute.yaml        |  4 +-
 ceilometer/templates/deployment-api.yaml           |  4 +-
 ceilometer/templates/deployment-central.yaml       |  4 +-
 ceilometer/templates/deployment-collector.yaml     |  4 +-
 ceilometer/templates/deployment-notification.yaml  |  4 +-
 ceilometer/templates/pod-rally-test.yaml           |  4 +-
 cinder/templates/configmap-etc.yaml                | 33 ++++++++-----
 .../cron-job-cinder-volume-usage-audit.yaml        |  4 +-
 cinder/templates/deployment-api.yaml               |  4 +-
 cinder/templates/deployment-backup.yaml            |  4 +-
 cinder/templates/deployment-scheduler.yaml         |  4 +-
 cinder/templates/deployment-volume.yaml            |  5 +-
 cinder/templates/job-backup-storage-init.yaml      |  4 +-
 cinder/templates/pod-rally-test.yaml               |  4 +-
 congress/templates/configmap-etc.yaml              | 15 +++---
 congress/templates/deployment-api.yaml             |  4 +-
 congress/templates/deployment-datasource.yaml      |  4 +-
 congress/templates/deployment-policy-engine.yaml   |  4 +-
 glance/templates/configmap-etc.yaml                | 26 ++++++----
 glance/templates/deployment-api.yaml               |  4 +-
 glance/templates/deployment-registry.yaml          |  4 +-
 glance/templates/pod-rally-test.yaml               |  4 +-
 heat/templates/configmap-etc.yaml                  | 21 +++++---
 heat/templates/cron-job-engine-cleaner.yaml        |  4 +-
 heat/templates/deployment-api.yaml                 |  4 +-
 heat/templates/deployment-cfn.yaml                 |  4 +-
 heat/templates/deployment-cloudwatch.yaml          |  4 +-
 heat/templates/deployment-engine.yaml              |  4 +-
 heat/templates/pod-rally-test.yaml                 |  4 +-
 horizon/templates/configmap-etc.yaml               | 10 ++--
 horizon/templates/deployment.yaml                  |  4 +-
 horizon/templates/job-db-sync.yaml                 |  4 +-
 ironic/templates/configmap-etc.yaml                | 16 +++---
 ironic/templates/deployment-api.yaml               |  4 +-
 ironic/templates/statefulset-conductor.yaml        |  4 +-
 keystone/templates/configmap-etc.yaml              | 29 ++++++-----
 keystone/templates/cron-job-credential-rotate.yaml |  4 +-
 keystone/templates/cron-job-fernet-rotate.yaml     |  4 +-
 keystone/templates/deployment-api.yaml             |  4 +-
 keystone/templates/job-credential-setup.yaml       |  8 +--
 keystone/templates/job-domain-manage.yaml          |  4 +-
 keystone/templates/job-fernet-setup.yaml           |  8 +--
 keystone/templates/pod-rally-test.yaml             |  4 +-
 magnum/templates/configmap-etc.yaml                | 15 +++---
 magnum/templates/deployment-api.yaml               |  4 +-
 magnum/templates/statefulset-conductor.yaml        |  4 +-
 mistral/templates/configmap-etc.yaml               | 18 ++++---
 mistral/templates/deployment-api.yaml              |  4 +-
 mistral/templates/deployment-executor.yaml         |  4 +-
 mistral/templates/pod-rally-test.yaml              |  4 +-
 mistral/templates/statefulset-engine.yaml          |  4 +-
 mistral/templates/statefulset-event-engine.yaml    |  4 +-
 mistral/values.yaml                                |  2 +-
 neutron/templates/configmap-etc.yaml               | 57 ++++++++++++++--------
 neutron/templates/daemonset-dhcp-agent.yaml        |  4 +-
 neutron/templates/daemonset-l3-agent.yaml          |  4 +-
 neutron/templates/daemonset-lb-agent.yaml          |  4 +-
 neutron/templates/daemonset-metadata-agent.yaml    |  4 +-
 neutron/templates/daemonset-ovs-agent.yaml         |  4 +-
 neutron/templates/daemonset-sriov-agent.yaml       |  4 +-
 neutron/templates/deployment-server.yaml           |  4 +-
 neutron/templates/pod-rally-test.yaml              |  4 +-
 nova/templates/configmap-etc.yaml                  | 34 ++++++++-----
 nova/templates/cron-job-cell-setup.yaml            |  4 +-
 nova/templates/cron-job-service-cleaner.yaml       |  4 +-
 nova/templates/daemonset-compute.yaml              |  4 +-
 nova/templates/deployment-api-metadata.yaml        |  4 +-
 nova/templates/deployment-api-osapi.yaml           |  4 +-
 nova/templates/deployment-conductor.yaml           |  4 +-
 nova/templates/deployment-consoleauth.yaml         |  4 +-
 nova/templates/deployment-novncproxy.yaml          |  4 +-
 nova/templates/deployment-placement.yaml           |  4 +-
 nova/templates/deployment-scheduler.yaml           |  4 +-
 nova/templates/deployment-spiceproxy.yaml          |  4 +-
 nova/templates/job-cell-setup.yaml                 |  4 +-
 nova/templates/pod-rally-test.yaml                 |  4 +-
 nova/templates/statefulset-compute-ironic.yaml     |  4 +-
 rally/templates/configmap-etc.yaml                 |  6 +--
 rally/templates/job-manage-db.yaml                 |  4 +-
 rally/templates/job-run-task.yaml                  |  4 +-
 senlin/templates/configmap-etc.yaml                | 18 ++++---
 senlin/templates/cron-job-engine-cleaner.yaml      |  4 +-
 senlin/templates/deployment-api.yaml               |  4 +-
 senlin/templates/deployment-engine.yaml            |  4 +-
 tempest/templates/configmap-etc.yaml               | 12 +++--
 tempest/templates/job-run-tests.yaml               |  4 +-
 89 files changed, 365 insertions(+), 294 deletions(-)

diff --git a/barbican/templates/configmap-etc.yaml b/barbican/templates/configmap-etc.yaml
index 18d5d37..b0bc113 100644
--- a/barbican/templates/configmap-etc.yaml
+++ b/barbican/templates/configmap-etc.yaml
@@ -86,15 +86,20 @@ limitations under the License.
 {{- end -}}
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: barbican-etc
-type: Opaque
 data:
-  barbican.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.barbican | b64enc }}
-  logging.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | b64enc }}
-  barbican-api-paste.ini: {{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | b64enc }}
-  api_audit_map.conf: {{ include "helm-toolkit.utils.to_ini" .Values.conf.audit_map | b64enc }}
-  policy.json: {{ toJson .Values.conf.policy | b64enc }}
-  barbican-api.ini: {{ include "helm-toolkit.utils.to_ini" .Values.conf.barbican_api | b64enc }}
+  barbican.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.barbican | indent 4 }}
+  logging.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | indent 4 }}
+  barbican-api-paste.ini: |
+{{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | indent 4 }}
+  api_audit_map.conf: |
+{{ include "helm-toolkit.utils.to_ini" .Values.conf.audit_map | indent 4 }}
+  policy.json: |
+{{ toJson .Values.conf.policy | indent 4 }}
+  barbican-api.ini: |
+{{ include "helm-toolkit.utils.to_ini" .Values.conf.barbican_api | indent 4 }}
 {{- end }}
diff --git a/barbican/templates/deployment-api.yaml b/barbican/templates/deployment-api.yaml
index 3187182..8270d36 100644
--- a/barbican/templates/deployment-api.yaml
+++ b/barbican/templates/deployment-api.yaml
@@ -109,8 +109,8 @@ spec:
         - name: etcbarbican
           emptyDir: {}
         - name: barbican-etc
-          secret:
-            secretName: barbican-etc
+          configMap:
+            name: barbican-etc
             defaultMode: 0444
         - name: barbican-bin
           configMap:
diff --git a/ceilometer/templates/configmap-etc.yaml b/ceilometer/templates/configmap-etc.yaml
index 17ddb1b..d7d9653 100644
--- a/ceilometer/templates/configmap-etc.yaml
+++ b/ceilometer/templates/configmap-etc.yaml
@@ -107,18 +107,25 @@ limitations under the License.
 
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: ceilometer-etc
-type: Opaque
 data:
-  rally_tests.yaml: {{ toYaml .Values.conf.rally_tests | b64enc }}
-  ceilometer.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.ceilometer | b64enc }}
-  api_paste.ini: {{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | b64enc }}
-  policy.json: {{ toJson .Values.conf.policy | b64enc }}
-  event_pipeline.yaml: {{ toYaml .Values.conf.event_pipeline | b64enc }}
-  pipeline.yaml: {{ toYaml .Values.conf.pipeline | b64enc }}
-  event_definitions.yaml: {{ toYaml .Values.conf.event_definitions | b64enc }}
-  gnocchi_resources.yaml: {{ toYaml .Values.conf.gnocchi_resources | b64enc }}
-{{ include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.wsgi_ceilometer "key" "wsgi-ceilometer.conf" "format" "Secret" ) | indent 2 }}
+  rally_tests.yaml: |
+{{ toYaml .Values.conf.rally_tests | indent 4 }}
+  ceilometer.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.ceilometer | indent 4 }}
+  api_paste.ini: |
+{{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | indent 4 }}
+  policy.json: |
+{{ toJson .Values.conf.policy | indent 4 }}
+  event_pipeline.yaml: |
+{{ toYaml .Values.conf.event_pipeline | indent 4 }}
+  pipeline.yaml: |
+{{ toYaml .Values.conf.pipeline | indent 4 }}
+  event_definitions.yaml: |
+{{ toYaml .Values.conf.event_definitions | indent 4 }}
+  gnocchi_resources.yaml: |
+{{ toYaml .Values.conf.gnocchi_resources | indent 4 }}
+{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.wsgi_ceilometer "key" "wsgi-ceilometer.conf" ) | indent 2 }}
 {{- end }}
diff --git a/ceilometer/templates/daemonset-compute.yaml b/ceilometer/templates/daemonset-compute.yaml
index 686572a..cf31840 100644
--- a/ceilometer/templates/daemonset-compute.yaml
+++ b/ceilometer/templates/daemonset-compute.yaml
@@ -111,8 +111,8 @@ spec:
         - name: pod-etc-ceilometer
           emptyDir: {}
         - name: ceilometer-etc
-          secret:
-            secretName: ceilometer-etc
+          configMap:
+            name: ceilometer-etc
             defaultMode: 0444
         - name: ceilometer-bin
           configMap:
diff --git a/ceilometer/templates/deployment-api.yaml b/ceilometer/templates/deployment-api.yaml
index 2a4c879..7ebc230 100644
--- a/ceilometer/templates/deployment-api.yaml
+++ b/ceilometer/templates/deployment-api.yaml
@@ -116,8 +116,8 @@ spec:
         - name: pod-etc-ceilometer
           emptyDir: {}
         - name: ceilometer-etc
-          secret:
-            secretName: ceilometer-etc
+          configMap:
+            name: ceilometer-etc
             defaultMode: 0444
         - name: ceilometer-bin
           configMap:
diff --git a/ceilometer/templates/deployment-central.yaml b/ceilometer/templates/deployment-central.yaml
index b7a597b..2fe7ed3 100644
--- a/ceilometer/templates/deployment-central.yaml
+++ b/ceilometer/templates/deployment-central.yaml
@@ -98,8 +98,8 @@ spec:
         - name: pod-etc-ceilometer
           emptyDir: {}
         - name: ceilometer-etc
-          secret:
-            secretName: ceilometer-etc
+          configMap:
+            name: ceilometer-etc
             defaultMode: 0444
         - name: ceilometer-bin
           configMap:
diff --git a/ceilometer/templates/deployment-collector.yaml b/ceilometer/templates/deployment-collector.yaml
index 07c761e..0fed543 100644
--- a/ceilometer/templates/deployment-collector.yaml
+++ b/ceilometer/templates/deployment-collector.yaml
@@ -98,8 +98,8 @@ spec:
         - name: pod-etc-ceilometer
           emptyDir: {}
         - name: ceilometer-etc
-          secret:
-            secretName: ceilometer-etc
+          configMap:
+            name: ceilometer-etc
             defaultMode: 0444
         - name: ceilometer-bin
           configMap:
diff --git a/ceilometer/templates/deployment-notification.yaml b/ceilometer/templates/deployment-notification.yaml
index 06fda3d..2eeb571 100644
--- a/ceilometer/templates/deployment-notification.yaml
+++ b/ceilometer/templates/deployment-notification.yaml
@@ -98,8 +98,8 @@ spec:
         - name: pod-etc-ceilometer
           emptyDir: {}
         - name: ceilometer-etc
-          secret:
-            secretName: ceilometer-etc
+          configMap:
+            name: ceilometer-etc
             defaultMode: 0444
         - name: ceilometer-bin
           configMap:
diff --git a/ceilometer/templates/pod-rally-test.yaml b/ceilometer/templates/pod-rally-test.yaml
index 9cbf2fd..f2dc212 100644
--- a/ceilometer/templates/pod-rally-test.yaml
+++ b/ceilometer/templates/pod-rally-test.yaml
@@ -87,8 +87,8 @@ spec:
 {{ if $mounts_tests.volumeMounts }}{{ toYaml $mounts_tests.volumeMounts | indent 8 }}{{ end }}
   volumes:
     - name: ceilometer-etc
-      secret:
-        secretName: ceilometer-etc
+      configMap:
+        name: ceilometer-etc
         defaultMode: 0444
     - name: ceilometer-bin
       configMap:
diff --git a/cinder/templates/configmap-etc.yaml b/cinder/templates/configmap-etc.yaml
index 88d9fa0..e116bde 100644
--- a/cinder/templates/configmap-etc.yaml
+++ b/cinder/templates/configmap-etc.yaml
@@ -108,24 +108,33 @@ limitations under the License.
 {{- end -}}
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: cinder-etc
-type: Opaque
 data:
-  rally_tests.yaml: {{ toYaml .Values.conf.rally_tests.tests | b64enc }}
-  cinder.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.cinder | b64enc }}
-  logging.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | b64enc }}
-  backends.conf: {{ include "helm-toolkit.utils.to_ini" .Values.conf.backends | b64enc }}
-  api-paste.ini: {{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | b64enc }}
-  policy.json: {{ toJson .Values.conf.policy | b64enc }}
-  cinder_sudoers: {{ $envAll.Values.conf.cinder_sudoers | b64enc }}
-  rootwrap.conf: {{ $envAll.Values.conf.rootwrap | b64enc }}
+  rally_tests.yaml: |
+{{ toYaml .Values.conf.rally_tests.tests | indent 4 }}
+  cinder.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.cinder | indent 4 }}
+  logging.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | indent 4 }}
+  backends.conf: |
+{{ include "helm-toolkit.utils.to_ini" .Values.conf.backends | indent 4 }}
+  api-paste.ini: |
+{{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | indent 4 }}
+  policy.json: |
+{{ toJson .Values.conf.policy | indent 4 }}
+  cinder_sudoers: |
+{{ $envAll.Values.conf.cinder_sudoers | indent 4 }}
+  rootwrap.conf: |
+{{ $envAll.Values.conf.rootwrap | indent 4 }}
 {{- range $key, $value := $envAll.Values.conf.rootwrap_filters }}
 {{- $filePrefix := replace "_" "-"  $key }}
-  {{ printf "%s.filters" $filePrefix }}: {{ $value.content | b64enc }}
+  {{ printf "%s.filters" $filePrefix }}: |
+{{ $value.content | indent 4 }}
 {{- end }}
 {{- if .Values.backup.external_ceph_rbd.enabled }}
-  external-backup-ceph.conf: {{ include "helm-toolkit.utils.to_ini" .Values.backup.external_ceph_rbd.conf | b64enc }}
+  external-backup-ceph.conf: |
+{{ include "helm-toolkit.utils.to_ini" .Values.backup.external_ceph_rbd.conf | indent 4 }}
 {{- end }}
 {{- end }}
diff --git a/cinder/templates/cron-job-cinder-volume-usage-audit.yaml b/cinder/templates/cron-job-cinder-volume-usage-audit.yaml
index 5d7ed3c..42ea01d 100644
--- a/cinder/templates/cron-job-cinder-volume-usage-audit.yaml
+++ b/cinder/templates/cron-job-cinder-volume-usage-audit.yaml
@@ -73,8 +73,8 @@ spec:
           - name: etccinder
             emptyDir: {}
           - name: cinder-etc
-            secret:
-              secretName: cinder-etc
+            configMap:
+              name: cinder-etc
               defaultMode: 0444
           - name: cinder-bin
             configMap:
diff --git a/cinder/templates/deployment-api.yaml b/cinder/templates/deployment-api.yaml
index 9213d34..629dc6b 100644
--- a/cinder/templates/deployment-api.yaml
+++ b/cinder/templates/deployment-api.yaml
@@ -120,8 +120,8 @@ spec:
             name: cinder-bin
             defaultMode: 0555
         - name: cinder-etc
-          secret:
-            secretName: cinder-etc
+          configMap:
+            name: cinder-etc
             defaultMode: 0444
         {{- if eq ( split "://" .Values.conf.cinder.coordination.backend_url )._0 "file" }}
         #NOTE (portdirect): this will need to be set to a shared mount amongst all cinder
diff --git a/cinder/templates/deployment-backup.yaml b/cinder/templates/deployment-backup.yaml
index d978b3a..181f132 100644
--- a/cinder/templates/deployment-backup.yaml
+++ b/cinder/templates/deployment-backup.yaml
@@ -197,8 +197,8 @@ spec:
 {{ if $mounts_cinder_backup.volumeMounts }}{{ toYaml $mounts_cinder_backup.volumeMounts | indent 12 }}{{ end }}
       volumes:
         - name: cinder-etc
-          secret:
-            secretName: cinder-etc
+          configMap:
+            name: cinder-etc
             defaultMode: 0444
         - name: cinder-bin
           configMap:
diff --git a/cinder/templates/deployment-scheduler.yaml b/cinder/templates/deployment-scheduler.yaml
index c60e13b..fb8512f 100644
--- a/cinder/templates/deployment-scheduler.yaml
+++ b/cinder/templates/deployment-scheduler.yaml
@@ -106,8 +106,8 @@ spec:
             name: cinder-bin
             defaultMode: 0555
         - name: cinder-etc
-          secret:
-            secretName: cinder-etc
+          configMap:
+            name: cinder-etc
             defaultMode: 0444
         {{- if eq ( split "://" .Values.conf.cinder.coordination.backend_url )._0 "file" }}
         #NOTE (portdirect): this will need to be set to a shared mount amongst all cinder
diff --git a/cinder/templates/deployment-volume.yaml b/cinder/templates/deployment-volume.yaml
index 46d13fd..7a1e7d5 100644
--- a/cinder/templates/deployment-volume.yaml
+++ b/cinder/templates/deployment-volume.yaml
@@ -161,9 +161,8 @@ spec:
             name: cinder-bin
             defaultMode: 0555
         - name: cinder-etc
-          secret:
-            secretName: cinder-etc
-            defaultMode: 0444
+          configMap:
+            name: cinder-etc
         {{- if include "cinder.utils.is_ceph_volume_configured" $envAll }}
         - name: etcceph
           emptyDir: {}
diff --git a/cinder/templates/job-backup-storage-init.yaml b/cinder/templates/job-backup-storage-init.yaml
index dcb756a..a2f7255 100644
--- a/cinder/templates/job-backup-storage-init.yaml
+++ b/cinder/templates/job-backup-storage-init.yaml
@@ -161,8 +161,8 @@ spec:
             defaultMode: 0444
         {{- if .Values.backup.external_ceph_rbd.enabled }}
         - name: cinder-etc
-          secret:
-            secretName: cinder-etc
+          configMap:
+            name: cinder-etc
             defaultMode: 0444
         {{- end }}
         {{- if empty .Values.conf.ceph.admin_keyring }}
diff --git a/cinder/templates/pod-rally-test.yaml b/cinder/templates/pod-rally-test.yaml
index f24e5aa..630e97b 100644
--- a/cinder/templates/pod-rally-test.yaml
+++ b/cinder/templates/pod-rally-test.yaml
@@ -89,8 +89,8 @@ spec:
 {{ if $mounts_tests.volumeMounts }}{{ toYaml $mounts_tests.volumeMounts | indent 8 }}{{ end }}
   volumes:
     - name: cinder-etc
-      secret:
-        secretName: cinder-etc
+      configMap:
+        name: cinder-etc
         defaultMode: 0444
     - name: cinder-bin
       configMap:
diff --git a/congress/templates/configmap-etc.yaml b/congress/templates/configmap-etc.yaml
index 4b4fd05..0996efd 100644
--- a/congress/templates/configmap-etc.yaml
+++ b/congress/templates/configmap-etc.yaml
@@ -77,13 +77,16 @@ limitations under the License.
 {{- end -}}
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: congress-etc
-type: Opaque
 data:
-  congress.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.congress | b64enc }}
-  logging.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | b64enc }}
-  api-paste.ini: {{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | b64enc }}
-  policy.json: {{ toJson .Values.conf.policy | b64enc }}
+  congress.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.congress | indent 4 }}
+  logging.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | indent 4 }}
+  api-paste.ini: |
+{{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | indent 4 }}
+  policy.json: |
+{{ toJson .Values.conf.policy | indent 4 }}
 {{- end }}
diff --git a/congress/templates/deployment-api.yaml b/congress/templates/deployment-api.yaml
index d635ce1..a78d9bc 100644
--- a/congress/templates/deployment-api.yaml
+++ b/congress/templates/deployment-api.yaml
@@ -91,7 +91,7 @@ spec:
             name: congress-bin
             defaultMode: 0777
         - name: congress-etc
-          secret:
-            secretName: congress-etc
+          configMap:
+            name: congress-etc
             defaultMode: 0444
 {{ end }}
diff --git a/congress/templates/deployment-datasource.yaml b/congress/templates/deployment-datasource.yaml
index 97cff78..5eeba89 100644
--- a/congress/templates/deployment-datasource.yaml
+++ b/congress/templates/deployment-datasource.yaml
@@ -85,8 +85,8 @@ spec:
             name: congress-bin
             defaultMode: 0555
         - name: congress-etc
-          secret:
-            secretName: congress-etc
+          configMap:
+            name: congress-etc
             defaultMode: 0444
 
 {{ end }}
diff --git a/congress/templates/deployment-policy-engine.yaml b/congress/templates/deployment-policy-engine.yaml
index 77c233f..d3f1bd8 100644
--- a/congress/templates/deployment-policy-engine.yaml
+++ b/congress/templates/deployment-policy-engine.yaml
@@ -85,8 +85,8 @@ spec:
             name: congress-bin
             defaultMode: 0555
         - name: congress-etc
-          secret:
-            secretName: congress-etc
+          configMap:
+            name: congress-etc
             defaultMode: 0444
 
 {{ end }}
diff --git a/glance/templates/configmap-etc.yaml b/glance/templates/configmap-etc.yaml
index 3c922db..c3d2f72 100644
--- a/glance/templates/configmap-etc.yaml
+++ b/glance/templates/configmap-etc.yaml
@@ -158,17 +158,23 @@ limitations under the License.
 {{- end -}}
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: glance-etc
-type: Opaque
 data:
-  rally_tests.yaml: {{ toYaml .Values.conf.rally_tests.tests | b64enc }}
-  glance-api.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.glance | b64enc }}
-  logging.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | b64enc }}
-  glance-api-paste.ini: {{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | b64enc }}
-  glance-registry.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.glance_registry | b64enc }}
-  glance-registry-paste.ini: {{ include "helm-toolkit.utils.to_ini" .Values.conf.paste_registry | b64enc }}
-  policy.json: {{ toJson .Values.conf.policy | b64enc }}
-{{- include "helm-toolkit.snippets.values_template_renderer" ( dict "envAll" $envAll "template" .Values.conf.swift_store "key" "swift-store.conf" "format" "Secret" ) | indent 2 }}
+  rally_tests.yaml: |
+{{ toYaml .Values.conf.rally_tests.tests | indent 4 }}
+  glance-api.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.glance | indent 4 }}
+  logging.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | indent 4 }}
+  glance-api-paste.ini: |
+{{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | indent 4 }}
+  glance-registry.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.glance_registry | indent 4 }}
+  glance-registry-paste.ini: |
+{{ include "helm-toolkit.utils.to_ini" .Values.conf.paste_registry | indent 4 }}
+  policy.json: |
+{{ toJson .Values.conf.policy | indent 4 }}
+{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.swift_store "key" "swift-store.conf") | indent 2 }}
 {{- end }}
diff --git a/glance/templates/deployment-api.yaml b/glance/templates/deployment-api.yaml
index 31316c7..634ceeb 100644
--- a/glance/templates/deployment-api.yaml
+++ b/glance/templates/deployment-api.yaml
@@ -159,8 +159,8 @@ spec:
             name: glance-bin
             defaultMode: 0555
         - name: glance-etc
-          secret:
-            secretName: glance-etc
+          configMap:
+            name: glance-etc
             defaultMode: 0444
 {{- if eq .Values.storage "pvc" }}
         - name: glance-images
diff --git a/glance/templates/deployment-registry.yaml b/glance/templates/deployment-registry.yaml
index cf296e7..5a4f614 100644
--- a/glance/templates/deployment-registry.yaml
+++ b/glance/templates/deployment-registry.yaml
@@ -106,8 +106,8 @@ spec:
             name: glance-bin
             defaultMode: 0555
         - name: glance-etc
-          secret:
-            secretName: glance-etc
+          configMap:
+            name: glance-etc
             defaultMode: 0444
 {{ if $mounts_glance_registry.volumes }}{{ toYaml $mounts_glance_registry.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/glance/templates/pod-rally-test.yaml b/glance/templates/pod-rally-test.yaml
index a0f992f..a807b83 100644
--- a/glance/templates/pod-rally-test.yaml
+++ b/glance/templates/pod-rally-test.yaml
@@ -88,8 +88,8 @@ spec:
 {{ if $mounts_tests.volumeMounts }}{{ toYaml $mounts_tests.volumeMounts | indent 8 }}{{ end }}
   volumes:
     - name: glance-etc
-      secret:
-        secretName: glance-etc
+      configMap:
+        name: glance-etc
         defaultMode: 0444
     - name: glance-bin
       configMap:
diff --git a/heat/templates/configmap-etc.yaml b/heat/templates/configmap-etc.yaml
index e89012d..7d56528 100644
--- a/heat/templates/configmap-etc.yaml
+++ b/heat/templates/configmap-etc.yaml
@@ -134,17 +134,22 @@ limitations under the License.
 {{- end -}}
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: heat-etc
-type: Opaque
 data:
-  rally_tests.yaml: {{ toYaml .Values.conf.rally_tests.tests | b64enc }}
-  heat.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.heat | b64enc }}
-  logging.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | b64enc }}
-  api-paste.ini: {{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | b64enc }}
-  policy.json: {{ toJson .Values.conf.policy | b64enc }}
+  rally_tests.yaml: |
+{{ toYaml .Values.conf.rally_tests.tests | indent 4 }}
+  heat.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.heat | indent 4 }}
+  logging.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | indent 4 }}
+  api-paste.ini: |
+{{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | indent 4 }}
+  policy.json: |
+{{  toJson .Values.conf.policy | indent 4 }}
 {{- range $key, $value := $envAll.Values.conf.rally_tests.templates }}
-  {{ printf "test_template_%d" $key }}: {{ $value.template | b64enc }}
+  {{ printf "test_template_%d" $key }}: |
+{{ $value.template | indent 4 }}
 {{- end }}
 {{- end }}
diff --git a/heat/templates/cron-job-engine-cleaner.yaml b/heat/templates/cron-job-engine-cleaner.yaml
index 986023f..77f8443 100644
--- a/heat/templates/cron-job-engine-cleaner.yaml
+++ b/heat/templates/cron-job-engine-cleaner.yaml
@@ -73,8 +73,8 @@ spec:
           - name: etcheat
             emptyDir: {}
           - name: heat-etc
-            secret:
-              secretName: heat-etc
+            configMap:
+              name: heat-etc
               defaultMode: 0444
           - name: heat-bin
             configMap:
diff --git a/heat/templates/deployment-api.yaml b/heat/templates/deployment-api.yaml
index e4799b2..be4c13f 100644
--- a/heat/templates/deployment-api.yaml
+++ b/heat/templates/deployment-api.yaml
@@ -106,8 +106,8 @@ spec:
             name: heat-bin
             defaultMode: 0555
         - name: heat-etc
-          secret:
-            secretName: heat-etc
+          configMap:
+            name: heat-etc
             defaultMode: 0444
 {{ if $mounts_heat_api.volumes }}{{ toYaml $mounts_heat_api.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/heat/templates/deployment-cfn.yaml b/heat/templates/deployment-cfn.yaml
index 81b165a..06fd278 100644
--- a/heat/templates/deployment-cfn.yaml
+++ b/heat/templates/deployment-cfn.yaml
@@ -106,8 +106,8 @@ spec:
             name: heat-bin
             defaultMode: 0555
         - name: heat-etc
-          secret:
-            secretName: heat-etc
+          configMap:
+            name: heat-etc
             defaultMode: 0444
 {{ if $mounts_heat_cfn.volumes }}{{ toYaml $mounts_heat_cfn.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/heat/templates/deployment-cloudwatch.yaml b/heat/templates/deployment-cloudwatch.yaml
index a93baba..e452b97 100644
--- a/heat/templates/deployment-cloudwatch.yaml
+++ b/heat/templates/deployment-cloudwatch.yaml
@@ -106,8 +106,8 @@ spec:
             name: heat-bin
             defaultMode: 0555
         - name: heat-etc
-          secret:
-            secretName: heat-etc
+          configMap:
+            name: heat-etc
             defaultMode: 0444
 {{ if $mounts_heat_cloudwatch.volumes }}{{ toYaml $mounts_heat_cloudwatch.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/heat/templates/deployment-engine.yaml b/heat/templates/deployment-engine.yaml
index a68939a..63d4543 100644
--- a/heat/templates/deployment-engine.yaml
+++ b/heat/templates/deployment-engine.yaml
@@ -104,8 +104,8 @@ spec:
             name: heat-bin
             defaultMode: 0555
         - name: heat-etc
-          secret:
-            secretName: heat-etc
+          configMap:
+            name: heat-etc
             defaultMode: 0444
 {{ if $mounts_heat_engine.volumes }}{{ toYaml $mounts_heat_engine.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/heat/templates/pod-rally-test.yaml b/heat/templates/pod-rally-test.yaml
index 2db25a3..3ddc6d3 100644
--- a/heat/templates/pod-rally-test.yaml
+++ b/heat/templates/pod-rally-test.yaml
@@ -95,8 +95,8 @@ spec:
 {{ if $mounts_tests.volumeMounts }}{{ toYaml $mounts_tests.volumeMounts | indent 8 }}{{ end }}
   volumes:
     - name: heat-etc
-      secret:
-        secretName: heat-etc
+      configMap:
+        name: heat-etc
         defaultMode: 0444
     - name: heat-bin
       configMap:
diff --git a/horizon/templates/configmap-etc.yaml b/horizon/templates/configmap-etc.yaml
index bfdfc18..dc695a1 100644
--- a/horizon/templates/configmap-etc.yaml
+++ b/horizon/templates/configmap-etc.yaml
@@ -18,14 +18,14 @@ limitations under the License.
 {{- $envAll := . }}
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: horizon-etc
-type: Opaque
 data:
-{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.horizon.apache "key" "horizon.conf" "format" "Secret" ) | indent 2 }}
-{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.horizon.local_settings.template "key" "local_settings" "format" "Secret" ) | indent 2 }}
+{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.horizon.apache "key" "horizon.conf") | indent 2 }}
+{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.horizon.local_settings.template "key" "local_settings") | indent 2 }}
 {{- range $key, $value := .Values.conf.horizon.policy }}
-  {{ printf "%s_policy.json" $key }}: {{ $value | toPrettyJson | b64enc }}
+  {{ printf "%s_policy.json" $key }}: |
+{{ $value | toPrettyJson | indent 4 }}
 {{- end }}
 {{- end }}
diff --git a/horizon/templates/deployment.yaml b/horizon/templates/deployment.yaml
index fe6b8da..2a76226 100644
--- a/horizon/templates/deployment.yaml
+++ b/horizon/templates/deployment.yaml
@@ -119,8 +119,8 @@ spec:
             name: horizon-bin
             defaultMode: 0555
         - name: horizon-etc
-          secret:
-            secretName: horizon-etc
+          configMap:
+            name: horizon-etc
             defaultMode: 0444
 {{ if $mounts_horizon.volumes }}{{ toYaml $mounts_horizon.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/horizon/templates/job-db-sync.yaml b/horizon/templates/job-db-sync.yaml
index 893428c..eefdb44 100644
--- a/horizon/templates/job-db-sync.yaml
+++ b/horizon/templates/job-db-sync.yaml
@@ -63,8 +63,8 @@ spec:
 {{ if $mounts_horizon_db_sync.volumeMounts }}{{ toYaml $mounts_horizon_db_sync.volumeMounts | indent 10 }}{{ end }}
       volumes:
       - name: horizon-etc
-        secret:
-          secretName: horizon-etc
+        configMap:
+          name: horizon-etc
           defaultMode: 0444
       - name: horizon-bin
         configMap:
diff --git a/ironic/templates/configmap-etc.yaml b/ironic/templates/configmap-etc.yaml
index 613e4bf..d0a62d5 100644
--- a/ironic/templates/configmap-etc.yaml
+++ b/ironic/templates/configmap-etc.yaml
@@ -198,14 +198,16 @@ limitations under the License.
 {{- end -}}
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: ironic-etc
-type: Opaque
 data:
-  ironic.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.ironic | b64enc }}
-  logging.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | b64enc }}
-  policy.json: {{  toJson .Values.conf.policy | b64enc }}
-{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.tftp_map_file "key" "tftp-map-file" "format" "Secret" ) | indent 2 }}
-{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.nginx "key" "nginx.conf" "format" "Secret" ) | indent 2 }}
+  ironic.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.ironic | indent 4 }}
+  logging.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | indent 4 }}
+  policy.json: |
+{{  toJson .Values.conf.policy | indent 4 }}
+{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.tftp_map_file "key" "tftp-map-file") | indent 2 }}
+{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.nginx "key" "nginx.conf") | indent 2 }}
 {{- end }}
diff --git a/ironic/templates/deployment-api.yaml b/ironic/templates/deployment-api.yaml
index 9498d58..2ef3631 100644
--- a/ironic/templates/deployment-api.yaml
+++ b/ironic/templates/deployment-api.yaml
@@ -132,8 +132,8 @@ spec:
             name: ironic-bin
             defaultMode: 0555
         - name: ironic-etc
-          secret:
-            secretName: ironic-etc
+          configMap:
+            name: ironic-etc
             defaultMode: 0444
         - name: pod-shared
           emptyDir: {}
diff --git a/ironic/templates/statefulset-conductor.yaml b/ironic/templates/statefulset-conductor.yaml
index 073690f..1f3036e 100644
--- a/ironic/templates/statefulset-conductor.yaml
+++ b/ironic/templates/statefulset-conductor.yaml
@@ -221,8 +221,8 @@ spec:
             name: ironic-bin
             defaultMode: 0555
         - name: ironic-etc
-          secret:
-            secretName: ironic-etc
+          configMap:
+            name: ironic-etc
             defaultMode: 0444
         - name: host-var-lib-ironic
           hostPath:
diff --git a/keystone/templates/configmap-etc.yaml b/keystone/templates/configmap-etc.yaml
index 5e7721b..edd97d9 100644
--- a/keystone/templates/configmap-etc.yaml
+++ b/keystone/templates/configmap-etc.yaml
@@ -43,21 +43,26 @@ limitations under the License.
 {{- end -}}
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: keystone-etc
-type: Opaque
 data:
-  rally_tests.yaml: {{ toYaml .Values.conf.rally_tests.tests | b64enc }}
-  keystone.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.keystone | b64enc }}
-  logging.conf: {{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.logging | b64enc }}
-  keystone-paste.ini: {{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | b64enc }}
-  policy.json: {{  toJson .Values.conf.policy | b64enc }}
-  ports.conf: ''
+  rally_tests.yaml: |
+{{ toYaml .Values.conf.rally_tests.tests | indent 4 }}
+  keystone.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.keystone | indent 4 }}
+  logging.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.logging | indent 4 }}
+  keystone-paste.ini: |
+{{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | indent 4 }}
+  policy.json: |
+{{  toJson .Values.conf.policy | indent 4 }}
+  ports.conf: ""
 {{- range $k, $v := .Values.conf.ks_domains }}
-  keystone.{{ $k }}.json: {{ toJson $v | b64enc }}
+  keystone.{{ $k }}.json: |
+{{ toJson $v | indent 4 }}
 {{- end }}
-{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.mpm_event "key" "mpm_event.conf" "format" "Secret" ) | indent 2 }}
-{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.wsgi_keystone "key" "wsgi-keystone.conf" "format" "Secret" ) | indent 2 }}
-{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.sso_callback_template "key" "sso_callback_template.html" "format" "Secret" ) | indent 2 }}
+{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.mpm_event "key" "mpm_event.conf") | indent 2 }}
+{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.wsgi_keystone "key" "wsgi-keystone.conf") | indent 2 }}
+{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.sso_callback_template "key" "sso_callback_template.html") | indent 2 }}
 {{- end }}
diff --git a/keystone/templates/cron-job-credential-rotate.yaml b/keystone/templates/cron-job-credential-rotate.yaml
index 9249675..2dcd47a 100644
--- a/keystone/templates/cron-job-credential-rotate.yaml
+++ b/keystone/templates/cron-job-credential-rotate.yaml
@@ -114,8 +114,8 @@ spec:
           - name: etckeystone
             emptyDir: {}
           - name: keystone-etc
-            secret:
-              secretName: keystone-etc
+            configMap:
+              name: keystone-etc
               defaultMode: 0444
           - name: keystone-bin
             configMap:
diff --git a/keystone/templates/cron-job-fernet-rotate.yaml b/keystone/templates/cron-job-fernet-rotate.yaml
index a61339e..353dfc2 100644
--- a/keystone/templates/cron-job-fernet-rotate.yaml
+++ b/keystone/templates/cron-job-fernet-rotate.yaml
@@ -113,8 +113,8 @@ spec:
           - name: etckeystone
             emptyDir: {}
           - name: keystone-etc
-            secret:
-              secretName: keystone-etc
+            configMap:
+              name: keystone-etc
               defaultMode: 0444
           - name: keystone-bin
             configMap:
diff --git a/keystone/templates/deployment-api.yaml b/keystone/templates/deployment-api.yaml
index c985cad..927ace0 100644
--- a/keystone/templates/deployment-api.yaml
+++ b/keystone/templates/deployment-api.yaml
@@ -147,8 +147,8 @@ spec:
       - name: run-apache
         emptyDir: {}
       - name: keystone-etc
-        secret:
-          secretName: keystone-etc
+        configMap:
+          name: keystone-etc
           defaultMode: 0444
       - name: keystone-bin
         configMap:
diff --git a/keystone/templates/job-credential-setup.yaml b/keystone/templates/job-credential-setup.yaml
index 4a212dc..b177b95 100644
--- a/keystone/templates/job-credential-setup.yaml
+++ b/keystone/templates/job-credential-setup.yaml
@@ -89,8 +89,6 @@ spec:
           volumeMounts:
           - name: etckeystone
             mountPath: /etc/keystone
-          - name: credential-keys
-            mountPath: {{ .Values.conf.keystone.credential.key_repository | quote }}
           - name: keystone-etc
             mountPath: /etc/keystone/keystone.conf
             subPath: keystone.conf
@@ -107,11 +105,9 @@ spec:
       volumes:
       - name: etckeystone
         emptyDir: {}
-      - name: credential-keys
-        emptyDir: {}
       - name: keystone-etc
-        secret:
-          secretName: keystone-etc
+        configMap:
+          name: keystone-etc
           defaultMode: 0444
       - name: keystone-bin
         configMap:
diff --git a/keystone/templates/job-domain-manage.yaml b/keystone/templates/job-domain-manage.yaml
index a2c8be8..df0bd03 100644
--- a/keystone/templates/job-domain-manage.yaml
+++ b/keystone/templates/job-domain-manage.yaml
@@ -103,8 +103,8 @@ spec:
         - name: etckeystonedomains
           emptyDir: {}
         - name: keystone-etc
-          secret:
-            secretName: keystone-etc
+          configMap:
+            name: keystone-etc
             defaultMode: 0444
         - name: keystone-bin
           configMap:
diff --git a/keystone/templates/job-fernet-setup.yaml b/keystone/templates/job-fernet-setup.yaml
index ebe9f21..58cd0a3 100644
--- a/keystone/templates/job-fernet-setup.yaml
+++ b/keystone/templates/job-fernet-setup.yaml
@@ -88,8 +88,6 @@ spec:
           volumeMounts:
           - name: etckeystone
             mountPath: /etc/keystone
-          - name: fernet-keys
-            mountPath: {{ .Values.conf.keystone.fernet_tokens.key_repository | quote }}
           - name: keystone-etc
             mountPath: /etc/keystone/keystone.conf
             subPath: keystone.conf
@@ -106,11 +104,9 @@ spec:
       volumes:
       - name: etckeystone
         emptyDir: {}
-      - name: fernet-keys
-        emptyDir: {}
       - name: keystone-etc
-        secret:
-          secretName: keystone-etc
+        configMap:
+          name: keystone-etc
           defaultMode: 0444
       - name: keystone-bin
         configMap:
diff --git a/keystone/templates/pod-rally-test.yaml b/keystone/templates/pod-rally-test.yaml
index d29bc0e..e68cef2 100644
--- a/keystone/templates/pod-rally-test.yaml
+++ b/keystone/templates/pod-rally-test.yaml
@@ -87,8 +87,8 @@ spec:
 {{ if $mounts_tests.volumeMounts }}{{ toYaml $mounts_tests.volumeMounts | indent 8 }}{{ end }}
   volumes:
     - name: keystone-etc
-      secret:
-        secretName: keystone-etc
+      configMap:
+        name: keystone-etc
         defaultMode: 0444
     - name: keystone-bin
       configMap:
diff --git a/magnum/templates/configmap-etc.yaml b/magnum/templates/configmap-etc.yaml
index 7796fa0..677abfc 100644
--- a/magnum/templates/configmap-etc.yaml
+++ b/magnum/templates/configmap-etc.yaml
@@ -87,13 +87,16 @@ limitations under the License.
 {{- end -}}
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: magnum-etc
-type: Opaque
 data:
-  magnum.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.magnum | b64enc }}
-  logging.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | b64enc }}
-  api-paste.ini: {{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | b64enc }}
-  policy.json: {{ toJson .Values.conf.policy | b64enc }}
+  magnum.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.magnum | indent 4 }}
+  logging.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | indent 4 }}
+  api-paste.ini: |
+{{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | indent 4 }}
+  policy.json: |
+{{ toJson .Values.conf.policy | indent 4 }}
 {{- end }}
diff --git a/magnum/templates/deployment-api.yaml b/magnum/templates/deployment-api.yaml
index 8015927..eb70e63 100644
--- a/magnum/templates/deployment-api.yaml
+++ b/magnum/templates/deployment-api.yaml
@@ -116,8 +116,8 @@ spec:
             name: magnum-bin
             defaultMode: 0555
         - name: magnum-etc
-          secret:
-            secretName: magnum-etc
+          configMap:
+            name: magnum-etc
             defaultMode: 0444
         - name: magnum-lock-path
           emptyDir: {}
diff --git a/magnum/templates/statefulset-conductor.yaml b/magnum/templates/statefulset-conductor.yaml
index 2c6f094..b8de04f 100644
--- a/magnum/templates/statefulset-conductor.yaml
+++ b/magnum/templates/statefulset-conductor.yaml
@@ -111,8 +111,8 @@ spec:
             name: magnum-bin
             defaultMode: 0555
         - name: magnum-etc
-          secret:
-            secretName: magnum-etc
+          configMap:
+            name: magnum-etc
             defaultMode: 0444
         - name: magnum-lock-path
           emptyDir: {}
diff --git a/mistral/templates/configmap-etc.yaml b/mistral/templates/configmap-etc.yaml
index 2e24c76..133cce3 100644
--- a/mistral/templates/configmap-etc.yaml
+++ b/mistral/templates/configmap-etc.yaml
@@ -77,16 +77,20 @@ limitations under the License.
 {{- end -}}
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: mistral-etc
-type: Opaque
 data:
-  rally_tests.yaml: {{ toYaml .Values.conf.rally_tests.tests | b64enc }}
-  mistral.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.mistral | b64enc }}
-  logging.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | b64enc }}
-  policy.json: {{ toJson .Values.conf.policy | b64enc }}
+  rally_tests.yaml: |
+{{ toYaml .Values.conf.rally_tests.tests | indent 4 }}
+  mistral.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.mistral | indent 4 }}
+  logging.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | indent 4 }}
+  policy.json: |
+{{  toJson .Values.conf.policy | indent 4 }}
 {{- range $key, $value := $envAll.Values.conf.rally_tests.templates }}
-  {{ printf "test_template_%d" $key }}: {{ $value.template | b64enc }}
+  {{ printf "test_template_%d" $key }}: |
+{{ $value.template | indent 4 }}
 {{- end }}
 {{- end }}
diff --git a/mistral/templates/deployment-api.yaml b/mistral/templates/deployment-api.yaml
index 0b5087c..d3b361d 100644
--- a/mistral/templates/deployment-api.yaml
+++ b/mistral/templates/deployment-api.yaml
@@ -102,8 +102,8 @@ spec:
             name: mistral-bin
             defaultMode: 0555
         - name: mistral-etc
-          secret:
-            secretName: mistral-etc
+          configMap:
+            name: mistral-etc
             defaultMode: 0444
 {{ if $mounts_mistral_api.volumes }}{{ toYaml $mounts_mistral_api.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/mistral/templates/deployment-executor.yaml b/mistral/templates/deployment-executor.yaml
index c66d22f..53234ab 100644
--- a/mistral/templates/deployment-executor.yaml
+++ b/mistral/templates/deployment-executor.yaml
@@ -84,8 +84,8 @@ spec:
             name: mistral-bin
             defaultMode: 0555
         - name: mistral-etc
-          secret:
-            secretName: mistral-etc
+          configMap:
+            name: mistral-etc
             defaultMode: 0444
 {{ if $mounts_mistral_executor.volumes }}{{ toYaml $mounts_mistral_executor.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/mistral/templates/pod-rally-test.yaml b/mistral/templates/pod-rally-test.yaml
index 70f85d9..2968c70 100644
--- a/mistral/templates/pod-rally-test.yaml
+++ b/mistral/templates/pod-rally-test.yaml
@@ -95,8 +95,8 @@ spec:
 {{ if $mounts_tests.volumeMounts }}{{ toYaml $mounts_tests.volumeMounts | indent 8 }}{{ end }}
   volumes:
     - name: mistral-etc
-      secret:
-        secretName: mistral-etc
+      configMap:
+        name: mistral-etc
         defaultMode: 0444
     - name: mistral-bin
       configMap:
diff --git a/mistral/templates/statefulset-engine.yaml b/mistral/templates/statefulset-engine.yaml
index 2b8e105..13def09 100644
--- a/mistral/templates/statefulset-engine.yaml
+++ b/mistral/templates/statefulset-engine.yaml
@@ -81,8 +81,8 @@ spec:
             name: mistral-bin
             defaultMode: 0555
         - name: mistral-etc
-          secret:
-            secretName: mistral-etc
+          configMap:
+            name: mistral-etc
             defaultMode: 0444
 {{ if $mounts_mistral_engine.volumes }}{{ toYaml $mounts_mistral_engine.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/mistral/templates/statefulset-event-engine.yaml b/mistral/templates/statefulset-event-engine.yaml
index bb2acb3..b9576c5 100644
--- a/mistral/templates/statefulset-event-engine.yaml
+++ b/mistral/templates/statefulset-event-engine.yaml
@@ -81,8 +81,8 @@ spec:
             name: mistral-bin
             defaultMode: 0555
         - name: mistral-etc
-          secret:
-            secretName: mistral-etc
+          configMap:
+            name: mistral-etc
             defaultMode: 0444
 {{ if $mounts_mistral_event_engine.volumes }}{{ toYaml $mounts_mistral_event_engine.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/mistral/values.yaml b/mistral/values.yaml
index d9e461e..9634c11 100644
--- a/mistral/values.yaml
+++ b/mistral/values.yaml
@@ -459,7 +459,7 @@ conf:
       port: null
       api_workers: 8
     coordination:
-      backend_url: ""
+      backend_url: null
     database:
       max_retries: -1
     keystone_authtoken:
diff --git a/neutron/templates/configmap-etc.yaml b/neutron/templates/configmap-etc.yaml
index 7293df0..ffed59c 100644
--- a/neutron/templates/configmap-etc.yaml
+++ b/neutron/templates/configmap-etc.yaml
@@ -171,32 +171,49 @@ just set it along with nova_metadata_host.
 {{- end -}}
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: {{ $configMapName }}
-type: Opaque
 data:
-  rally_tests.yaml: {{ toYaml $envAll.Values.conf.rally_tests.tests | b64enc }}
-  api-paste.ini: {{ include "helm-toolkit.utils.to_ini" $envAll.Values.conf.paste | b64enc }}
-  policy.json: {{ toJson $envAll.Values.conf.policy | b64enc }}
-  neutron.conf: {{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.neutron | b64enc }}
-  logging.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | b64enc }}
-  dhcp_agent.ini: {{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.dhcp_agent | b64enc }}
-  l3_agent.ini: {{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.l3_agent | b64enc }}
-  metadata_agent.ini: {{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.metadata_agent | b64enc }}
-  metering_agent.ini: {{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.metering_agent | b64enc }}
-  ml2_conf.ini: {{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.plugins.ml2_conf | b64enc }}
-  ml2_conf_sriov.ini: {{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.plugins.ml2_conf_sriov | b64enc }}
-  macvtap_agent.ini: {{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.plugins.macvtap_agent | b64enc }}
-  linuxbridge_agent.ini: {{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.plugins.linuxbridge_agent | b64enc }}
-  openvswitch_agent.ini: {{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.plugins.openvswitch_agent | b64enc }}
-  sriov_agent.ini: {{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.plugins.sriov_agent | b64enc }}
+  rally_tests.yaml: |
+{{ toYaml $envAll.Values.conf.rally_tests.tests | indent 4 }}
+  api-paste.ini: |
+{{ include "helm-toolkit.utils.to_ini" $envAll.Values.conf.paste | indent 4 }}
+  policy.json: |
+{{ toJson $envAll.Values.conf.policy | indent 4 }}
+  neutron.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.neutron | indent 4 }}
+  logging.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | indent 4 }}
+  dhcp_agent.ini: |
+{{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.dhcp_agent | indent 4 }}
+  l3_agent.ini: |
+{{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.l3_agent | indent 4 }}
+  metadata_agent.ini: |
+{{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.metadata_agent | indent 4 }}
+  metering_agent.ini: |
+{{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.metering_agent | indent 4 }}
+  ml2_conf.ini: |
+{{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.plugins.ml2_conf | indent 4 }}
+  ml2_conf_sriov.ini: |
+{{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.plugins.ml2_conf_sriov | indent 4 }}
+  macvtap_agent.ini: |
+{{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.plugins.macvtap_agent | indent 4 }}
+  linuxbridge_agent.ini: |
+{{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.plugins.linuxbridge_agent | indent 4 }}
+  openvswitch_agent.ini: |
+{{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.plugins.openvswitch_agent | indent 4 }}
+  sriov_agent.ini: |
+{{ include "helm-toolkit.utils.to_oslo_conf" $envAll.Values.conf.plugins.sriov_agent | indent 4 }}
   dnsmasq.conf: ""
-  neutron_sudoers: {{ $envAll.Values.conf.neutron_sudoers | b64enc }}
-  rootwrap.conf: {{ $envAll.Values.conf.rootwrap | b64enc }}
+  neutron_sudoers: |
+{{ $envAll.Values.conf.neutron_sudoers | indent 4 }}
+  rootwrap.conf: |
+{{ $envAll.Values.conf.rootwrap | indent 4 }}
 {{- range $key, $value := $envAll.Values.conf.rootwrap_filters }}
 {{- $filePrefix := replace "_" "-"  $key }}
-  {{ printf "%s.filters" $filePrefix }}: {{ $value.content | b64enc }}
+  {{ printf "%s.filters" $filePrefix }}: |
+{{ $value.content | indent 4 }}
 {{- end }}
 {{- end }}
 {{- end }}
diff --git a/neutron/templates/daemonset-dhcp-agent.yaml b/neutron/templates/daemonset-dhcp-agent.yaml
index e7d84aa..a45c08b 100644
--- a/neutron/templates/daemonset-dhcp-agent.yaml
+++ b/neutron/templates/daemonset-dhcp-agent.yaml
@@ -137,8 +137,8 @@ spec:
             name: neutron-bin
             defaultMode: 0555
         - name: neutron-etc
-          secret:
-            secretName: {{ $configMapName }}
+          configMap:
+            name: {{ $configMapName }}
             defaultMode: 0444
         - name: socket
           hostPath:
diff --git a/neutron/templates/daemonset-l3-agent.yaml b/neutron/templates/daemonset-l3-agent.yaml
index ac0a3f7..d91bef3 100644
--- a/neutron/templates/daemonset-l3-agent.yaml
+++ b/neutron/templates/daemonset-l3-agent.yaml
@@ -136,8 +136,8 @@ spec:
             name: neutron-bin
             defaultMode: 0555
         - name: neutron-etc
-          secret:
-            secretName: {{ $configMapName }}
+          configMap:
+            name: {{ $configMapName }}
             defaultMode: 0444
         - name: libmodules
           hostPath:
diff --git a/neutron/templates/daemonset-lb-agent.yaml b/neutron/templates/daemonset-lb-agent.yaml
index 7730ab4..bf209a6 100644
--- a/neutron/templates/daemonset-lb-agent.yaml
+++ b/neutron/templates/daemonset-lb-agent.yaml
@@ -196,8 +196,8 @@ spec:
             name: neutron-bin
             defaultMode: 0555
         - name: neutron-etc
-          secret:
-            secretName: {{ $configMapName }}
+          configMap:
+            name: {{ $configMapName }}
             defaultMode: 0444
         - name: run
           hostPath:
diff --git a/neutron/templates/daemonset-metadata-agent.yaml b/neutron/templates/daemonset-metadata-agent.yaml
index 2a24aa6..c5e65d8 100644
--- a/neutron/templates/daemonset-metadata-agent.yaml
+++ b/neutron/templates/daemonset-metadata-agent.yaml
@@ -150,8 +150,8 @@ spec:
             name: neutron-bin
             defaultMode: 0555
         - name: neutron-etc
-          secret:
-            secretName: {{ $configMapName }}
+          configMap:
+            name: {{ $configMapName }}
             defaultMode: 0444
         - name: socket
           hostPath:
diff --git a/neutron/templates/daemonset-ovs-agent.yaml b/neutron/templates/daemonset-ovs-agent.yaml
index 3ca1ca3..04a29e1 100644
--- a/neutron/templates/daemonset-ovs-agent.yaml
+++ b/neutron/templates/daemonset-ovs-agent.yaml
@@ -200,8 +200,8 @@ spec:
             name: neutron-bin
             defaultMode: 0555
         - name: neutron-etc
-          secret:
-            secretName: {{ $configMapName }}
+          configMap:
+            name: {{ $configMapName }}
             defaultMode: 0444
         - name: run
           hostPath:
diff --git a/neutron/templates/daemonset-sriov-agent.yaml b/neutron/templates/daemonset-sriov-agent.yaml
index 6130817..e1213a7 100644
--- a/neutron/templates/daemonset-sriov-agent.yaml
+++ b/neutron/templates/daemonset-sriov-agent.yaml
@@ -173,8 +173,8 @@ spec:
             name: neutron-bin
             defaultMode: 0555
         - name: neutron-etc
-          secret:
-            secretName: {{ $configMapName }}
+          configMap:
+            name: {{ $configMapName }}
             defaultMode: 0444
         - name: run
           hostPath:
diff --git a/neutron/templates/deployment-server.yaml b/neutron/templates/deployment-server.yaml
index 31dec3d..628937b 100644
--- a/neutron/templates/deployment-server.yaml
+++ b/neutron/templates/deployment-server.yaml
@@ -115,8 +115,8 @@ spec:
             name: neutron-bin
             defaultMode: 0555
         - name: neutron-etc
-          secret:
-            secretName: neutron-etc
+          configMap:
+            name: neutron-etc
             defaultMode: 0444
 {{ if $mounts_neutron_server.volumes }}{{ toYaml $mounts_neutron_server.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/neutron/templates/pod-rally-test.yaml b/neutron/templates/pod-rally-test.yaml
index 15633d1..16ced6e 100644
--- a/neutron/templates/pod-rally-test.yaml
+++ b/neutron/templates/pod-rally-test.yaml
@@ -90,8 +90,8 @@ spec:
 {{ if $mounts_tests.volumeMounts }}{{ toYaml $mounts_tests.volumeMounts | indent 8 }}{{ end }}
   volumes:
     - name: neutron-etc
-      secret:
-        secretName: neutron-etc
+      configMap:
+        name: neutron-etc
         defaultMode: 0444
     - name: neutron-bin
       configMap:
diff --git a/nova/templates/configmap-etc.yaml b/nova/templates/configmap-etc.yaml
index d1a55f5..cbd69ee 100644
--- a/nova/templates/configmap-etc.yaml
+++ b/nova/templates/configmap-etc.yaml
@@ -214,26 +214,34 @@ limitations under the License.
 {{- end -}}
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: {{ $configMapName }}
-type: Opaque
 data:
-  rally_tests.yaml: {{ toYaml .Values.conf.rally_tests.tests | b64enc }}
-  api-paste.ini: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.paste | b64enc }}
-  policy.yaml: {{ toYaml .Values.conf.policy | b64enc }}
-  nova_sudoers: {{ $envAll.Values.conf.nova_sudoers | b64enc }}
-  rootwrap.conf: {{ .Values.conf.rootwrap | b64enc }}
+  rally_tests.yaml: |
+{{ toYaml .Values.conf.rally_tests.tests | indent 4 }}
+  api-paste.ini: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.paste | indent 4 }}
+  policy.yaml: |
+{{ toYaml .Values.conf.policy | indent 4 }}
+  nova_sudoers: |
+{{ $envAll.Values.conf.nova_sudoers | indent 4 }}
+  rootwrap.conf: |
+{{- .Values.conf.rootwrap | indent 4 }}
 {{- range $key, $value := $envAll.Values.conf.rootwrap_filters }}
 {{- $filePrefix := replace "_" "-"  $key }}
-  {{ printf "%s.filters" $filePrefix }}: {{ $value.content | b64enc }}
+  {{ printf "%s.filters" $filePrefix }}: |
+{{ $value.content | indent 4 }}
 {{- end }}
-  nova.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.nova | b64enc }}
-  logging.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | b64enc }}
-  nova-ironic.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.nova_ironic | b64enc }}
-{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.wsgi_placement "key" "wsgi-nova-placement.conf" "format" "Secret" ) | indent 2 }}
+  nova.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.nova | indent 4 }}
+  logging.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | indent 4 }}
+  nova-ironic.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.nova_ironic | indent 4 }}
+{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.wsgi_placement "key" "wsgi-nova-placement.conf" ) | indent 2 }}
 # FIXME(portdirect): why is this file suffixed .sh?
-{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.ssh "key" "ssh-config.sh" "format" "Secret" ) | indent 2 }}
+{{- include "helm-toolkit.snippets.values_template_renderer" (dict "envAll" $envAll "template" .Values.conf.ssh "key" "ssh-config.sh" ) | indent 2 }}
 {{- end }}
 {{- end }}
 {{- if .Values.manifests.configmap_etc }}
diff --git a/nova/templates/cron-job-cell-setup.yaml b/nova/templates/cron-job-cell-setup.yaml
index 66df4df..dd71748 100644
--- a/nova/templates/cron-job-cell-setup.yaml
+++ b/nova/templates/cron-job-cell-setup.yaml
@@ -76,8 +76,8 @@ spec:
             - name: etcnova
               emptyDir: {}
             - name: nova-etc
-              secret:
-                secretName: nova-etc
+              configMap:
+                name: nova-etc
                 defaultMode: 0444
             - name: nova-bin
               configMap:
diff --git a/nova/templates/cron-job-service-cleaner.yaml b/nova/templates/cron-job-service-cleaner.yaml
index 889cfc4..db36ae6 100644
--- a/nova/templates/cron-job-service-cleaner.yaml
+++ b/nova/templates/cron-job-service-cleaner.yaml
@@ -65,8 +65,8 @@ spec:
           - name: etcnova
             emptyDir: {}
           - name: nova-etc
-            secret:
-              secretName: nova-etc
+            configMap:
+              name: nova-etc
               defaultMode: 0444
           - name: nova-bin
             configMap:
diff --git a/nova/templates/daemonset-compute.yaml b/nova/templates/daemonset-compute.yaml
index 850f0b0..ae681bb 100644
--- a/nova/templates/daemonset-compute.yaml
+++ b/nova/templates/daemonset-compute.yaml
@@ -292,8 +292,8 @@ spec:
             name: nova-bin
             defaultMode: 0555
         - name: nova-etc
-          secret:
-            secretName: {{ $configMapName }}
+          configMap:
+            name: {{ $configMapName }}
             defaultMode: 0444
         {{- if .Values.conf.ceph.enabled }}
         - name: etcceph
diff --git a/nova/templates/deployment-api-metadata.yaml b/nova/templates/deployment-api-metadata.yaml
index dfc9947..5b55950 100644
--- a/nova/templates/deployment-api-metadata.yaml
+++ b/nova/templates/deployment-api-metadata.yaml
@@ -155,8 +155,8 @@ spec:
             name: nova-bin
             defaultMode: 0555
         - name: nova-etc
-          secret:
-            secretName: nova-etc
+          configMap:
+            name: nova-etc
             defaultMode: 0444
         - name: pod-shared
           emptyDir: {}
diff --git a/nova/templates/deployment-api-osapi.yaml b/nova/templates/deployment-api-osapi.yaml
index f0aa805..263d3a8 100644
--- a/nova/templates/deployment-api-osapi.yaml
+++ b/nova/templates/deployment-api-osapi.yaml
@@ -102,8 +102,8 @@ spec:
             name: nova-bin
             defaultMode: 0555
         - name: nova-etc
-          secret:
-            secretName: nova-etc
+          configMap:
+            name: nova-etc
             defaultMode: 0444
 {{ if $mounts_nova_api_osapi.volumes}}{{ toYaml $mounts_nova_api_osapi.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/nova/templates/deployment-conductor.yaml b/nova/templates/deployment-conductor.yaml
index 7bf0841..36c326f 100644
--- a/nova/templates/deployment-conductor.yaml
+++ b/nova/templates/deployment-conductor.yaml
@@ -84,8 +84,8 @@ spec:
             name: nova-bin
             defaultMode: 0555
         - name: nova-etc
-          secret:
-            secretName: nova-etc
+          configMap:
+            name: nova-etc
             defaultMode: 0444
 {{ if $mounts_nova_conductor.volumes }}{{ toYaml $mounts_nova_conductor.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/nova/templates/deployment-consoleauth.yaml b/nova/templates/deployment-consoleauth.yaml
index 10de3fc..2a42917 100644
--- a/nova/templates/deployment-consoleauth.yaml
+++ b/nova/templates/deployment-consoleauth.yaml
@@ -84,8 +84,8 @@ spec:
             name: nova-bin
             defaultMode: 0555
         - name: nova-etc
-          secret:
-            secretName: nova-etc
+          configMap:
+            name: nova-etc
             defaultMode: 0444
 {{ if $mounts_nova_consoleauth.volumes }}{{ toYaml $mounts_nova_consoleauth.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/nova/templates/deployment-novncproxy.yaml b/nova/templates/deployment-novncproxy.yaml
index 85e0e11..992650b 100644
--- a/nova/templates/deployment-novncproxy.yaml
+++ b/nova/templates/deployment-novncproxy.yaml
@@ -122,8 +122,8 @@ spec:
             name: nova-bin
             defaultMode: 0555
         - name: nova-etc
-          secret:
-            secretName: nova-etc
+          configMap:
+            name: nova-etc
             defaultMode: 0444
         - name: pod-usr-share-novnc
           emptyDir: {}
diff --git a/nova/templates/deployment-placement.yaml b/nova/templates/deployment-placement.yaml
index db62d75..4838f93 100644
--- a/nova/templates/deployment-placement.yaml
+++ b/nova/templates/deployment-placement.yaml
@@ -111,8 +111,8 @@ spec:
             name: nova-bin
             defaultMode: 0555
         - name: nova-etc
-          secret:
-            secretName: nova-etc
+          configMap:
+            name: nova-etc
             defaultMode: 0444
 {{- if $mounts_nova_placement.volumes }}{{ toYaml $mounts_nova_placement.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/nova/templates/deployment-scheduler.yaml b/nova/templates/deployment-scheduler.yaml
index dc8b0d3..018566f 100644
--- a/nova/templates/deployment-scheduler.yaml
+++ b/nova/templates/deployment-scheduler.yaml
@@ -84,8 +84,8 @@ spec:
             name: nova-bin
             defaultMode: 0555
         - name: nova-etc
-          secret:
-            secretName: nova-etc
+          configMap:
+            name: nova-etc
             defaultMode: 0444
 {{ if $mounts_nova_scheduler.volumes }}{{ toYaml $mounts_nova_scheduler.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/nova/templates/deployment-spiceproxy.yaml b/nova/templates/deployment-spiceproxy.yaml
index 65fa900..351d54f 100644
--- a/nova/templates/deployment-spiceproxy.yaml
+++ b/nova/templates/deployment-spiceproxy.yaml
@@ -122,8 +122,8 @@ spec:
             name: nova-bin
             defaultMode: 0555
         - name: nova-etc
-          secret:
-            secretName: nova-etc
+          configMap:
+            name: nova-etc
             defaultMode: 0444
         - name: pod-usr-share-spice-html5
           emptyDir: {}
diff --git a/nova/templates/job-cell-setup.yaml b/nova/templates/job-cell-setup.yaml
index 4ad9509..ffabd1f 100644
--- a/nova/templates/job-cell-setup.yaml
+++ b/nova/templates/job-cell-setup.yaml
@@ -81,8 +81,8 @@ spec:
         - name: etcnova
           emptyDir: {}
         - name: nova-etc
-          secret:
-            secretName: nova-etc
+          configMap:
+            name: nova-etc
             defaultMode: 0444
         - name: nova-bin
           configMap:
diff --git a/nova/templates/pod-rally-test.yaml b/nova/templates/pod-rally-test.yaml
index f7b2e1b..05e0558 100644
--- a/nova/templates/pod-rally-test.yaml
+++ b/nova/templates/pod-rally-test.yaml
@@ -89,8 +89,8 @@ spec:
 {{ if $mounts_tests.volumeMounts }}{{ toYaml $mounts_tests.volumeMounts | indent 8 }}{{ end }}
   volumes:
     - name: nova-etc
-      secret:
-        secretName: nova-etc
+      configMap:
+        name: nova-etc
         defaultMode: 0444
     - name: nova-bin
       configMap:
diff --git a/nova/templates/statefulset-compute-ironic.yaml b/nova/templates/statefulset-compute-ironic.yaml
index eb39dd8..19cd9fe 100644
--- a/nova/templates/statefulset-compute-ironic.yaml
+++ b/nova/templates/statefulset-compute-ironic.yaml
@@ -97,8 +97,8 @@ spec:
             name: nova-bin
             defaultMode: 0555
         - name: nova-etc
-          secret:
-            secretName: nova-etc
+          configMap:
+            name: nova-etc
             defaultMode: 0444
         - name: varlibironic
           hostPath:
diff --git a/rally/templates/configmap-etc.yaml b/rally/templates/configmap-etc.yaml
index 2bbe5fe..afa651f 100644
--- a/rally/templates/configmap-etc.yaml
+++ b/rally/templates/configmap-etc.yaml
@@ -54,10 +54,10 @@ limitations under the License.
 
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: rally-etc
-type: Opaque
 data:
-  rally.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.rally | b64enc }}
+  rally.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.rally | indent 4 }}
 {{- end }}
diff --git a/rally/templates/job-manage-db.yaml b/rally/templates/job-manage-db.yaml
index 7510e69..82a2188 100644
--- a/rally/templates/job-manage-db.yaml
+++ b/rally/templates/job-manage-db.yaml
@@ -59,8 +59,8 @@ spec:
         - name: etcrally
           emptyDir: {}
         - name: rally-etc
-          secret:
-            secretName: rally-etc
+          configMap:
+            name: rally-etc
             defaultMode: 0444
         - name: rally-bin
           configMap:
diff --git a/rally/templates/job-run-task.yaml b/rally/templates/job-run-task.yaml
index ed14757..2b091d5 100644
--- a/rally/templates/job-run-task.yaml
+++ b/rally/templates/job-run-task.yaml
@@ -86,8 +86,8 @@ spec:
         - name: etcrally
           emptyDir: {}
         - name: rally-etc
-          secret:
-            secretName: rally-etc
+          configMap:
+            name: rally-etc
             defaultMode: 0444
         - name: rally-tasks
           configMap:
diff --git a/senlin/templates/configmap-etc.yaml b/senlin/templates/configmap-etc.yaml
index 027b468..cf9477a 100644
--- a/senlin/templates/configmap-etc.yaml
+++ b/senlin/templates/configmap-etc.yaml
@@ -97,14 +97,18 @@ limitations under the License.
 {{- end -}}
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: senlin-etc
-type: Opaque
 data:
-  rally_tests.yaml: {{ toYaml .Values.conf.rally_tests.tests | b64enc }}
-  senlin.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.senlin | b64enc }}
-  logging.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | b64enc }}
-  api-paste.ini: {{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | b64enc }}
-  policy.json: {{  toJson .Values.conf.policy | b64enc }}
+  rally_tests.yaml: |
+{{ toYaml .Values.conf.rally_tests.tests | indent 4 }}
+  senlin.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.senlin | indent 4 }}
+  logging.conf: |
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.logging | indent 4 }}
+  api-paste.ini: |
+{{ include "helm-toolkit.utils.to_ini" .Values.conf.paste | indent 4 }}
+  policy.json: |
+{{  toJson .Values.conf.policy | indent 4 }}
 {{- end }}
diff --git a/senlin/templates/cron-job-engine-cleaner.yaml b/senlin/templates/cron-job-engine-cleaner.yaml
index 18883e4..ce0dd3b 100644
--- a/senlin/templates/cron-job-engine-cleaner.yaml
+++ b/senlin/templates/cron-job-engine-cleaner.yaml
@@ -76,8 +76,8 @@ spec:
           - name: etcsenlin
             emptyDir: {}
           - name: senlin-etc
-            secret:
-              secretName: senlin-etc
+            configMap:
+              name: senlin-etc
               defaultMode: 0444
           - name: senlin-bin
             configMap:
diff --git a/senlin/templates/deployment-api.yaml b/senlin/templates/deployment-api.yaml
index 93eb9b8..0eca3d3 100644
--- a/senlin/templates/deployment-api.yaml
+++ b/senlin/templates/deployment-api.yaml
@@ -114,8 +114,8 @@ spec:
             name: senlin-bin
             defaultMode: 0555
         - name: senlin-etc
-          secret:
-            secretName: senlin-etc
+          configMap:
+            name: senlin-etc
             defaultMode: 0444
 {{ if $mounts_senlin_api.volumes }}{{ toYaml $mounts_senlin_api.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/senlin/templates/deployment-engine.yaml b/senlin/templates/deployment-engine.yaml
index fd54515..5c0714b 100644
--- a/senlin/templates/deployment-engine.yaml
+++ b/senlin/templates/deployment-engine.yaml
@@ -87,8 +87,8 @@ spec:
             name: senlin-bin
             defaultMode: 0555
         - name: senlin-etc
-          secret:
-            secretName: senlin-etc
+          configMap:
+            name: senlin-etc
             defaultMode: 0444
 {{ if $mounts_senlin_engine.volumes }}{{ toYaml $mounts_senlin_engine.volumes | indent 8 }}{{ end }}
 {{- end }}
diff --git a/tempest/templates/configmap-etc.yaml b/tempest/templates/configmap-etc.yaml
index 855c460..bd44108 100644
--- a/tempest/templates/configmap-etc.yaml
+++ b/tempest/templates/configmap-etc.yaml
@@ -43,16 +43,18 @@ limitations under the License.
 
 ---
 apiVersion: v1
-kind: Secret
+kind: ConfigMap
 metadata:
   name: tempest-etc
-type: Opaque
 data:
-  tempest.conf: {{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.tempest | b64enc }}
+  tempest.conf: |+
+{{ include "helm-toolkit.utils.to_oslo_conf" .Values.conf.tempest | indent 4 }}
 {{ if not (empty .Values.conf.blacklist) }}
-  test-blacklist: {{ include "tempest.utils.to_regex_file" .Values.conf.blacklist | b64enc }}
+  test-blacklist: |+
+{{ include "tempest.utils.to_regex_file" .Values.conf.blacklist | indent 4 }}
 {{ end }}
 {{ if not (empty .Values.conf.whitelist) }}
-  test-whitelist: {{ include "tempest.utils.to_regex_file" .Values.conf.whitelist | b64enc }}
+  test-whitelist: |+
+{{ include "tempest.utils.to_regex_file" .Values.conf.whitelist | indent 4 }}
 {{ end }}
 {{- end }}
diff --git a/tempest/templates/job-run-tests.yaml b/tempest/templates/job-run-tests.yaml
index 6f29e1a..1865aa5 100644
--- a/tempest/templates/job-run-tests.yaml
+++ b/tempest/templates/job-run-tests.yaml
@@ -91,8 +91,8 @@ spec:
         - name: etctempest
           emptyDir: {}
         - name: tempest-etc
-          secret:
-            secretName: tempest-etc
+          configMap:
+            name: tempest-etc
             defaultMode: 0444
         - name: tempest-bin
           configMap:
-- 
1.8.3.1

